bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/SunSpecHandlerFactory.java

```diff
@@ -69,14 +63,16 @@ public SunSpecHandlerFactory(@Reference ModbusManager manager) {
 
     @Override
     public boolean supportsThingType(ThingTypeUID thingTypeUID) {
-        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+        return SUPPORTED_THING_TYPES_UIDS.containsValue(thingTypeUID);
     }
 
     @Override
     protected @Nullable ThingHandler createHandler(Thing thing) {
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
 
-        if (thingTypeUID.equals(THING_TYPE_INVERTER_SINGLE_PHASE)) {
+        if (thingTypeUID.equals(THING_TYPE_INVERTER_SINGLE_PHASE)
+                || thingTypeUID.equals(THING_TYPE_INVERTER_SPLIT_PHASE)
+                || thingTypeUID.equals(THING_TYPE_INVERTER_THREE_PHASE)) {
```

Comment:
Would this work?
```suggestion
        if (supportsThingType(thingTypeUID)) {
```

Reply:
I have the 4th PR ready wich will add support for smart meters. But they use a different handler class, so this if will have an `else if ...` block with other thing types. That's the reason why this isn't a simple map lookup at the moment..